goreq_test.go

```diff
@@ -1065,6 +1066,98 @@ func TestRequest(t *testing.T) {
 
 		})
 
+		g.Describe("TLS Proxy", func() {
```

Comment:
This test seem a bit overcomplicated. You just need to verify that the header gets passed to the proxy. No need to implement a whole proxy in the test.

Reply:
Since I could not think of a simple test, I am wondering if there is no problem as a test.

Reply:
@kilisima what I'm saying is that you don't need to implement a whole proxy. You can safely remove the TLS server, and then do the assertion in the httptest server. No need to `dial` to a new server. You just need to assert that the headers arrive in the `proxyCh` which you're already doing.

Reply:
I removed the TLS server from the test case and simplified the test case.
Thank you. I appreciate it.

Reply:
@kilisima thank you for doing that. After reviewing the PR again I realize that there's no need for a `Describe("TLS Proxy"`) block as we're not testing anything related to TLS here. I think it's best if we remove that `describe` and we move the test to the `Proxy` describe block and re-use the http test server that's being used there which makes sense. That way everything will be consistent and even simple. 

Sorry to bring this up this late, It's that I just realized about that.

Reply:
Fixed. Please check.